src/components/input/inputNumber/inputNumber.tsx

```diff
@@ -0,0 +1,204 @@
+        }
+
+        // ensure the new value is within the min and max range
+        newValue = Math.max(min, Math.min(max, newValue));
+        setValue(newValue.toString());
+    };
+
+    const handleDecrement = () => {
+        const parsedValue = Number(unmaskedValue ?? 0);
+
+        // if the current value is less than the min, don't decrement
+        if (parsedValue < min) {
+            setValue(parsedValue.toString());
+        }
+
+        // decrement value by the step
+        let newValue = parsedValue - step;
+
+        // if the value is not a multiple of the step,
+        // decrement to the biggest multiple of the step that is less than the value
+        if (parsedValue % step !== 0) {
+            newValue = Math.floor(parsedValue / step) * step;
+        }
+
+        // ensure the new value is within the min and max range
+        newValue = Math.max(min, Math.min(max, newValue));
+        setValue(newValue.toString());
+    };
+
+    return (
+        <InputContainer
+            className={classNames('relative', containerClassName)}
+            {...otherContainerProps}
+            isDisabled={isDisabled}
+        >
+            {!isDisabled && (
+                <Button
+                    size="sm"
+                    variant="tertiary"
+                    onClick={handleDecrement}
+                    iconLeft={IconType.REMOVE}
+                    className="ml-2 shrink-0"
+                />
+            )}
+            <input
+                ref={numberMaskRef}
+                min={min}
+                max={max}
+                step={step}
+                value={isFocused ? maskedValue : suffixedValue}
+                onBlur={handleBlur}
+                onFocus={handleFocus}
+                disabled={isDisabled}
+                inputMode="numeric"
+                onKeyDown={handleKeyDown}
+                className={classNames('text-center spin-buttons:appearance-none', inputClassName)}
+                {...otherInputProps}
+            />
+            {!isDisabled && (
+                <Button
+                    size="sm"
+                    variant="tertiary"
+                    iconLeft={IconType.ADD}
+                    onClick={handleIncrement}
+                    className="mr-2 shrink-0"
+                />
+            )}
+        </InputContainer>
+    );
+};
+
+/**
+ * Parses the input values and returns an object with the parsed values.
+ *
+ * @param value - The value of the number input. Defaults to '0'.
+ * @param min - The minimum value of the number input. Defaults to Number.MIN_SAFE_INTEGER.
+ * @param max - The maximum value of the number input. Defaults to Number.MAX_SAFE_INTEGER.
+ * @returns  An object with the parsed values.
+ */
+function parseInputs(
+    value: IInputNumberProps['value'] = '',
+    step: IInputNumberProps['step'] = 1,
+    min: IInputNumberProps['min'] = Number.MIN_SAFE_INTEGER,
+    max: IInputNumberProps['max'] = Number.MAX_SAFE_INTEGER,
```

Comment:
Let's specify these props on the `IInputNumberProps` interface so that we can document their default values. As we're going to specify these props there, I would suggest to only accept the number type and we can remove much of this logic

Reply:
A bit iffy about this for a couple of reasons. 
1. It's a major departure from the native html input API; the more familiar we keep the API, the easier the DX and adoption I think
2. `value` has to be at least a `string` (or `string` | `number`) for it to make sense with the IMask `onChange` type which is returning a string. Forcing `value` into a `number` means the burden falls on the consumer to map cast the controlled `onChange` value into a `number` when setting the state.

Reply:
I agree for the `value` property, the comment is related to the `step`, `min` and `max` properties. I mean, we're already setting some default values which are not set by default, I think we should be fine in only setting these props as number.